File: 프로그래머스/Level2/고득점 Kit/타겟넘버-DFSBFS.py
answer = 0

def dfs(numbers, target, cnt, total) :
    if cnt == len(numbers) :
        if total == target :
            global answer
            answer +=1
        return
    
    dfs(numbers, target, cnt+1, total+numbers[cnt])
    dfs(numbers, target, cnt+1, total-numbers[cnt])
    
    return

def solution(numbers, target):
    global answer
    
    dfs(numbers, target, 1, -numbers[0])
    dfs(numbers, target, 1, numbers[0])

    return answer

File: Source Code/c++/프로그래머스/Level2/고득점 Kit/타겟넘버-DFSBFS.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int visited[21] = {0,};
vector<int> v;
int answer = 0;

void init(){
    for(int i=0; i<21; i++){
        visited[i] = 0;
    }
    v.clear();
    return;
}

void dfs(vector<int> numbers, int target, int sz, int cnt, int total){
    
    if(cnt==sz){
        if(total==target){
            answer++;
        }
        return;
    }
    
    dfs(numbers, target, sz, cnt+1, total+numbers[cnt]);
    dfs(numbers, target, sz, cnt+1, total-numbers[cnt]);
    
    return;
}

int solution(vector<int> numbers, int target) {
    int sz = numbers.size();
    int total = numbers[0];
    
    dfs(numbers, target, sz, 1, total);
    dfs(numbers, target, sz, 1, -total);
    
    return answer;
}